xbmc/addons/AddonRepos.h

```diff
@@ -0,0 +1,188 @@
+/*
+ *  Copyright (C) 2005-2020 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#pragma once
+
+#include "AddonDatabase.h"
+
+#include <map>
+#include <vector>
+
+namespace ADDON
+{
+/**
+ * Struct - RepoInfo
+ */
+struct RepoInfo
+{
+  std::string m_repoId;
+  std::string m_origin;
+};
+
+class CAddonMgr;
+class CRepository;
+class IAddon;
+
+/**
+ * Class - CAddonRepos
+ */
+class CAddonRepos
+{
+public:
+  CAddonRepos(const CAddonMgr& addonMgr) : m_addonMgr(addonMgr){}; // ctor
+
+  /*!
+   * \brief Load the map of all available addon versions in any installed repository
+   * \param database reference to the database to load addons from
+   */
+  void LoadAddonsFromDatabase(const CAddonDatabase& database);
+
+  /*!
+   * \brief Load the map of all available versions of an addonId in any installed repository
+   * \param database reference to the database to load addons from
+   * \param addonId the addon id we want to retrieve versions for
+   */
+  void LoadAddonsFromDatabase(const CAddonDatabase& database, const std::string& addonId);
+
+  /*!
+   * \brief Build the list of addons to be updated depending on defined rules
+   * \param installed vector of all addons installed on the system that are
+   *        checked for an update
+   * \param [out] updates list of addon versions that are going to be installed
```

Comment:
If you put a `[out]` it must be on `\param` without space.

Here and also on other places